OSDOCS-12135: Documented the 4.15.35 z-stream RNs (#82535)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-15-35_{context}"]
=== RHSA-2024:7179 - {product-title} 4.15.35 bug fix and security update

Issued: 02 October 2024

{product-title} release 4.15.35, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2024:7179[RHSA-2024:7179] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHSA-2024:7182[RHSA-2024:7182] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.35 --pullspecs
----

[id="ocp-4-15-35-bug-fixes_{context}"]
==== Bug fixes

> * Previously, when you upgraded an {product-title} cluster from 4.14 to 4.15, the `vCenterCluster` parameter was not populated with a value in the `use-connection-form.ts` configuration file. As a result, the {vmw-full} GUI did not display {vmw-full} vCenter information. With this release, an update to the Infrastructure custom resource (CR) ensures it has that the GUI checks the 'cloud-provider-config' ConfigMap for the `vCenterCluster` value. (link:https://issues.redhat.com/browse/OCPBUGS-42144[*OCPBUGS-42144*])

Reviewer: ```suggestion
* Previously, when you upgraded an {product-title} cluster from 4.14 to 4.15, the `vCenterCluster` parameter was not populated with a value in the `use-connection-form.ts` configuration file. As a result, the {vmw-full} GUI did not display {vmw-full} vCenter information. With this release, an update to the Infrastructure custom resource (CR) ensures it has that the GUI checks the `cloud-provider-config` ConfigMap for the `vCenterCluster` value. (link:https://issues.redhat.com/browse/OCPBUGS-42144[*OCPBUGS-42144*])
```

> * Previously, deploying a self-managed private hosted cluster on {aws-first} fails because the `bootstrap-kubeconfig` file uses an incorrect `kube-apiserver` port. As a result, the {aws-short} instances are provisioned cannot join the hosted cluster as nodes. With this release, the issue is fixed so this issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-42214[*OCPBUGS-42214*])

Reviewer: ```suggestion
* Previously, deploying a self-managed private hosted cluster on {aws-first} fails because the `bootstrap-kubeconfig` file uses an incorrect `kube-apiserver` port. As a result, the {aws-short} instances are provisioned but cannot join the hosted cluster as nodes. With this release, the issue is fixed so this issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-42214[*OCPBUGS-42214*])
```

> * Previously, when the hosted cluster `controllerAvailabilityPolicy` was set to `SingleReplica`, `podAntiAffinity` on networking components blocked the availability of the components. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42020[*OCPBUGS-42020*])

Reviewer: I don't know what noun to suggest, but ideally the stuff in backticks should have nouns following.

Author: Checked out existing docs and "property" is mentioned. Updated.

> * Previously, adding IPv6 support for user-provisioned installation platforms caused an issue with naming {rh-openstack-first} resources, especially when you run two user-provisioned installation clusters on the same {rh-openstack-first} platform. This happened because the two clusters share the same names for network, subnets, and router resources. With this release, all the resources names for a cluster remain unique for that cluster so no interfere occurs. (link:https://issues.redhat.com/browse/OCPBUGS-42011[*OCPBUGS-42011*])

Reviewer: ```suggestion
* Previously, adding IPv6 support for user-provisioned installation platforms caused an issue with naming {rh-openstack-first} resources, especially when you run two user-provisioned installation clusters on the same {rh-openstack} platform. This happened because the two clusters share the same names for network, subnets, and router resources. With this release, all the resources names for a cluster remain unique for that cluster so no interfere occurs. (link:https://issues.redhat.com/browse/OCPBUGS-42011[*OCPBUGS-42011*])
```

> * Previously, when you configured a hosted cluster to use an identity provider (IdP) that has either an `http` or `https` endpoint, the IdP hostname does not resolve when sent through the proxy. With this release, a DNS lookup operation checks the IdP before IdP traffic is send through a proxy, so that IdPs with hostnames can only be resolved by the data plane and verified by the Control Plane Operator (CPO). (link:https://issues.redhat.com/browse/OCPBUGS-41373[*OCPBUGS-41373*])

Reviewer: ```suggestion
* Previously, when you configured a hosted cluster to use an identity provider (IdP) that has either an `http` or `https` endpoint, the IdP hostname did not resolve when sent through the proxy. With this release, a DNS lookup operation checks the IdP before IdP traffic is sent through a proxy, so that IdPs with hostnames can only be resolved by the data plane and verified by the Control Plane Operator (CPO). (link:https://issues.redhat.com/browse/OCPBUGS-41373[*OCPBUGS-41373*])
```

> * Previously, a group ID was not added to the `/etc/group` within the container when the `spec.securityContext.runAsGroup` attribute was set in the Pod resource. With this release, this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-41245[*OCPBUGS-41245*])

Reviewer: (or just say "set in the pod.")

```suggestion
* Previously, a group ID was not added to the `/etc/group` within the container when the `spec.securityContext.runAsGroup` attribute was set in the `Pod` resource. With this release, this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-41245[*OCPBUGS-41245*])
```

> * Previously, the order of an Ansible playbook was modified to run before the metadata.json file was created, which caused issues with older versions of Ansible. With this release, the playbook is more tolerant of missing files and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-39288[*OCPBUGS-39288*])

Reviewer: ```suggestion
* Previously, the order of an Ansible playbook was modified to run before the `metadata.json` file was created, which caused issues with older versions of Ansible. With this release, the playbook is more tolerant of missing files and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-39288[*OCPBUGS-39288*])
```

> * Previously, dynamic plugins using PatternFly 4 were referencing variables that are not available in {product-title} 4.15 and later. This was causing contrast issues for ACM in dark mode. With this update, older chart styles are now available to support PatternFly 4 charts used by dynamic plugins. (link:https://issues.redhat.com/browse/OCPBUGS-38537[*OCPBUGS-38537*])

Reviewer: ```suggestion
* Previously, dynamic plugins using PatternFly 4 referenced variables that are not available in {product-title} 4.15 and later. This caused contrast issues for {rh-rhacm-first} in dark mode. With this update, older chart styles are now available to support PatternFly 4 charts used by dynamic plugins. (link:https://issues.redhat.com/browse/OCPBUGS-38537[*OCPBUGS-38537*])
```

> * Previously, proxying for IDP communication occurred in the Konnectivity agent. By the time traffic reached Konnectivity, its protocol and hostname was no longer available. As a consequence, proxying was not done correctly for the OAUTH server pod. It did not distinguish between protocols that require proxying (http/s) and protocols that do not (ldap://). In addition, it did not honor the `no_proxy` variable that is configured in the `HostedCluster.spec.configuration.proxy` spec. With this release, you can configure the proxy on the Konnectivity sidecar of the OAUTH server so that traffic is routed appropriately, honoring your `no_proxy` settings. As a result, the OAUTH server can communicate properly with identity providers when a proxy is configured for the hosted cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38058[*OCPBUGS-38058*])

Reviewer: ```suggestion
* Previously, proxying for identity provider (IdP) communication occurred in the Konnectivity agent. By the time traffic reached Konnectivity, its protocol and hostname was no longer available. As a consequence, proxying was not done correctly for the OAUTH server pod. It did not distinguish between protocols that require proxying (`http` or `https`) and protocols that do not (LDAP). In addition, it did not honor the `no_proxy` variable that is configured in the `HostedCluster.spec.configuration.proxy` spec. With this release, you can configure the proxy on the Konnectivity sidecar of the OAUTH server so that traffic is routed appropriately, honoring your `no_proxy` settings. As a result, the OAUTH server can communicate properly with IdPs when a proxy is configured for the hosted cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38058[*OCPBUGS-38058*])
```

> * Previously, the `assisted-installer` did not reload new data from the `assisted-service` when the `assisted-installer` checked control plane nodes for readiness and a conflict existed with a write operation from the `assisted-installer-controller`. This conflict prevented the `assisted-installer` from detecting a node that was marked by the `assisted-installer-controller` as `Ready` because the `assisted-installer` relied on older information. With this release, the `assisted-installer` can receive the newest information from the `assisted-service`, so that it the `assisted-installer` can accurately detect the status of each node. (link:https://issues.redhat.com/browse/OCPBUGS-38003[*OCPBUGS-38003*])

Reviewer: Checked against AI docs: https://www.redhat.com/en/blog/how-to-use-the-openshift-assisted-installer
```suggestion
* Previously, the {ai-full} did not reload new data from the Assisted Service when the {ai-full} checked control plane nodes for readiness and a conflict existed with a write operation from the {ai-full} controller. This conflict prevented the {ai-full} from detecting a node that was marked by the {ai-full} controller as `Ready` because the {ai-full} relied on older information. With this release, the {ai-full} can receive the newest information from the Assisted Service, so that it the {ai-full} can accurately detect the status of each node. (link:https://issues.redhat.com/browse/OCPBUGS-38003[*OCPBUGS-38003*])
```
